src/Games/NexusMods.Games.BethesdaGameStudios/BethesdaLoadoutSynchronizer.cs

```diff
@@ -0,0 +1,70 @@
+
+using NexusMods.DataModel.Abstractions;
+using NexusMods.DataModel.Games;
+using NexusMods.DataModel.Loadouts;
+using NexusMods.DataModel.Loadouts.Mods;
+using NexusMods.DataModel.LoadoutSynchronizer;
+
+namespace NexusMods.Games.BethesdaGameStudios;
+
+public class BethesdaLoadoutSynchronizer : ALoadoutSynchronizer
+{
+    public BethesdaLoadoutSynchronizer(IServiceProvider provider) : base(provider) { }
+
+    public override async Task<Loadout> Manage(GameInstallation installation)
+    {
+        var loadout = await base.Manage(installation);
+        return FixupLoadout(loadout);
+    }
+
+    /// <summary>
+    /// Patches the loadout to ensure it has the required metadata mod and plugin order file.
+    /// </summary>
+    /// <param name="loadout"></param>
+    /// <returns></returns>
+    protected virtual Loadout FixupLoadout(Loadout loadout)
```

Comment:
This could probably be abstracted, the issue is that there's currently no way to keep the user from getting the loadout into an invalid state. What happens if they delete the plugings.txt file? And if they do, how is that inserted back in? The idea is that we can call this at any time to make sure we haven't deleted critical game files. 

I think a better idea would be to have some sort of validation rule system in the new design, like a way to reject Events that would cause an invalid state. So we could have rules like "you can never delete the plugins.txt file from the loadout".